### DISCUSSION TITLE
Linked Data Taxonomies in Pubs

### DISCUSSION THREAD

**Discussion by gabestein** [knowledgefutures/pubpub#1386] [Ideas]
A very interesting suggestion from [Twitter](https://twitter.com/brimwats/status/1388502096639447042):

> any thoughts about allowing the use of linked data taxonomies attached to a Pub or (even neater) within the text? It would allow someone to (f.ex) say "turtle island" and link to geonames for what is also called N. America, or to use library headings within their pub.

The closest we've come to thinking about this is allowing more flexible metadata models to be attached to Pubs, generally. But I'm curious if anyone else has thoughts about (or examples of) using LD taxonomies in CMSs.

**Comment by isTravis** [comment]
@coahearn and I just had a call on Friday with a group who has an existing book they'd also like to publish as a stand-alone linked data Collection on Collector(pulling out all the entities and relationships from the book and then formalizing that as a linked dataset). If there was a bridge that let them both publish the book and dataset, and then connect the two in meaningful ways, I think they'd be thrilled.

**Reply by coahearn** [reply]
yah, pretty crazy that these two ideas came through back-to-back!

**Reply by brimwats1** [reply]
This is so cool and I would love to learn more about the book. It definitely gets at what I'm interested in and talked about below

**Comment by brimwats1** [comment]
Hello! As the person that originally reached out, I think it's a great idea 😉

It would also establish pubpub far in advance of just about everyone else out there as I documented on the wikidata mailing list about a year ago (but still relevant!):

------------------------------------

I was wondering if anyone was familiar with a semantic/linked data capable content management system or blog that has autofill or nanotation capabilities. What I mean by that is, say I'm writing a blog post about Paris, I'm looking for something that would autofill linked data 'under the hood' by either a dropdown (a la Omeka's [Value Suggest](https://omeka.org/s/modules/ValueSuggest/)), a autofill (a la wikidata/pedia) or something that creates semantic blog tags.

I've seen a (very) bleeding-edge technology/proof of concept called [nanotation](http://kidehen.blogspot.com/2014/07/nanotation.html) that looks about right, but might be completely different then what I actually want, which is to find something that incorporates linked data, autofills URIs, and works like a blog/content management system.

So far I've explored

-   **Recogito** ([https://recogito.pelagios.org/](https://recogito.pelagios.org/)) is lovely but focused on annotating images/maps/preexisting items.
-   **Catma** ([https://catma.de/](https://catma.de/)) is lovely looking but builds off preexisting texts, not creating new texts (i.e. you'd have to write the text and then annotate it all.). It seems to be a Voyant on steroids. Nonetheless if I could combine Recogito and Catma, that'd be neat. The same program (? project?) also puts out forText ([https://fortext.net/](https://fortext.net/)), which i just include here as it's also nice.
-   **[dokie.li](http://dokie.li/)** ([https://dokie.li/](https://dokie.li/)) This seems the closest, as it's focused on article publishing, annotations and social interactions, but unfortunately, setting up a Solid Server remains quite the technical hurdle for me
-   **Atomgraph** ([https://atomgraph.com/](https://atomgraph.com/)) is knowledge graph oriented and installed upon previously-existing data, not focused on content management. Gephi on steroids.
-   **Webanno** ([https://webanno.github.io/webanno/](https://webanno.github.io/webanno/)) which is specifically targeted at linguistically annotating the internet, not really creating content.
-   **Wikibase**: A heavily modified wikibase might be what I'm left with. In this scenario I'd make a Mediawiki, turn it into Wikibase, and kinda hack a blog out of it. Less than satisfying but would work if needed.
-   I also tried **wiki.js** (SUCH A NICE INTERFACE, but it doesn't support linked data yet) and **OntoWiki** (which looks like it also builds off a preexisting knowledge graph)
-   **Anthologize**: ([https://anthologize.org/](https://anthologize.org/)) also looks very close as a wordpress plugin but it is not linked\-data specific so I didn't explore ways to make it so.
-   I've also explored [**wordpress**](https://wordpress.org/plugins/wp-linked-data/) and Drupal plugins ([one](https://www.drupal.org/project/ldp), [two](https://www.drupal.org/project/linked_data), [three](https://www.drupal.org/project/ldt)) that are all obsolete or not maintained anymore
    

My longterm goal with this is to create semantic libguides and blogs. I really do think semantic libguides are NEARLY possible—maybe an API that pulls knowledge graphs along and wikidata visualizations, along with some blog-type software... I think it could be done, and I have some bits and pieces of it, but not quite the whole sandwich (so to speak).

**Comment by brimwats1** [comment]
Here are some examples of what this might look like:

* http://blog.sparna.fr/2020/02/20/semantic-markdown/
* https://medium.com/@kidehen/using-nanotation-within-markdown-github-41d3d425aac2


@ozekik has a proposal for markdown-ld here: https://github.com/ozekik/markdown-ld
It looks like @mbakeranalecta has been working on one as well: https://github.com/mbakeranalecta/sam

**Comment by micahvandegrift** [comment]
I'd love this capability too. What is described in this thread is what I thought the structured data feature in a citation would accomplish. We're doing this inelegantly using citations, links, and footnotes currently on Fermentology.

**Comment by gabestein** [comment]
Hello! This has come up in a couple conversations recently, and so I wanted to revisit the (extremely helpful, thank you!) comments here and run a thought by you, @micahvandegrift and @brimwats.

Basically, I'm wondering if the following train of thought (apologies for the vagueness of it) might get at what you're looking for: what if we expanded “connections” to be “taxonomies” and allowed people to add more arbitrary structures/taxonomies/vocabs? My thinking basically went like this: I want to be able to add connections when I create links. Wait aren’t connections just links, then? Oh no, they add the crossref relationship taxonomy to the link. Wait, what if you could add other taxonomies in the same way?

I think that leads me to two questions:
- How close is that (assume "link" is very general, and auto-complete, and ability to add taxonomies, etc.) to what you're envisioning from a UX perspective? I took a look at several of the examples you cited, including nanotation, and the most straightforward UX, and the one I'm mostly thinking about, is CATMA.
- Given that, any thoughts on the best way to represent and expose those annotations under-the-hood in a released Pub, which effectively targets HTML? I'm pretty new to this world, but my sense is that although microdata is the most straightforward way to do so if we're using the link metaphor, it's not particularly in vogue.

**Reply by katycroffbell** [reply]
Late to the party, but FWIW, I'd love to be able to add my own taxonomy. I'm working on a collection that has multiple pairs of pubs, one is a Full Results section, and one is a Summary. None of the current Connection labels fits that relationship (not a reply, or a comment, or supplement), but I'd really love to have them connected.

**Reply by gabestein** [reply]
Unfortunately, we're somewhat limited in terms of the Connections taxonomy, because it maps back to Crossref (and soon, Datacite) for relationship deposits. And we also don't want it to become too greenfield, because that makes it hard to compare what people mean by Connections across different PubPub communities.

But we are thinking we could expand to include some other existing relationship taxonomies (e.g. the [FaBiO Ontology](https://sparontologies.github.io/fabio/current/fabio.html)) and then alias them back to Crossref/Datacite ontologies. We'd love to hear suggestions there for types you'd like us to add.

**Reply by katycroffbell** [reply]
That all makes sense. For my purposes, I was hoping to link summary and full results together. For now I made the summary the main pub, and the full results a "Supplement."

To keep it somewhat constrained, you could add the components of a book/volume, like Abstract and/or Summary, Methods, Results, Discussion, Conclusions...those are pretty evergreen and I could see different sections/chapters being individual pubs (which is more or less what we are doing).

**Reply by gabestein** [reply]
Gotcha. Probably silly question, and I may not understand what you're going for, but: is using a Book collection not sufficient for accomplishing this, or could it be enhanced in some way to make what you're trying to accomplish easier? Using a Book Collection automatically adds a dropdown to the top left of the pub with everything in order, and similarly, a read next link in the pub footer to the next Pub. Seems simpler/more useful than using Connections, which are really made for defining relationships between things rather than defining types of things, if that makes sense. In fact, we sort of resist defining types for things at all directly, since we're trying not to make people conform to specific shapes of objects (instead, we like to try to guess types of object based on their other properties at the time they're needed for systems that need types to make them legible, e.g. Crossref).

**Reply by katycroffbell** [reply]
Ah, yeah, I see what you mean for the book. In my case, it's a report with sections focused on different regions of the world. So there's a Regional Summary pub for Africa, and a Results section for Africa, a Regional Summary pub for Asia, and a Results section for Asia, etc.. I also have data tables for each region. I'd like the pubs that are related to each region linked to each other. It's not really a linear "book" but i did use that format for the collection for the way the citation turns out.

**Reply by gabestein** [reply]
Gotcha. That's very cool. I might tag in @dawit-tegbaru or @coahearn to see what they recommend for this case. You can certainly use tags to organize things along multiple dimensions (a la [Fermentology](https://fermentology.pubpub.org/)), but I get that's not entirely sufficient here. And neither the crossref nor datacite taxonomies offer us much. What we might consider is adding a "Summary" relationship type that corresponds to Crossref's weirdly-named "Derivation" type. For the data tables, they recommend Supplement, which is probably what I'd recommend as well.

More on crossref relationship types: https://www.crossref.org/documentation/schema-library/markup-guide-metadata-segments/relationships/

**Reply by Dawit-Tegbaru** [reply]
It seems you and Katy covered the bases pretty well here. Agree that setting the connection type to 'Supplement' for pubs containing datasets makes sense. Katy, it's also good practice to cite your results and dataset pubs in the summary reports (if you're not already doing so). 

Looks like tags are already set up for regions, methods, summaries, results, data — this makes sense for navigation/discovery. 👍

The only point I'd make further RE: Crossref's connection types is consider what makes sense to your community (not just machine readers). Generally, folks are accustomed to declarations like: comment, reply, supplement, translation, etc. 

If we can accommodate the linkage with 'Derivation', consider also 'Basis', which may be more user-friendly. 

Let us know which of the [existing declarations](https://www.crossref.org/documentation/schema-library/markup-guide-metadata-segments/relationships/) fit best for your use case, as Gabe notee it may be possible for us to adjust the available connection types for you. In case it helps, we've outlined an example two-way connection scenario below using the Basis and Supplement declarations:  

Basis: 
Full results isBasisFor Summary  
Summary isBasedOn Full Results

Supplement: 
Geo Data Tables: Africa isSupplementTo Summary
Summary isSupplementedBy Geo Data Tables: Africa

**Reply by katycroffbell** [reply]
Right now, I'm using Supplement for everything but basis would definitely be more appropriate for the relationship between the summaries & results. I also have a references page (haven't made it into a pub), as well as research protocols (which could use "Related material"). I think those are the most relevant for my purposes from the Crossref list.

**Comment by mbakeranalecta** [comment]
My take on this has always been that is very hard to address this problem in a stable universal way on the front end, but actually quite easy to address it in a local way on the back end. If you can establish the semantics of a link text on the back end, you can transform that into any external semantics you need on the front end. I advocate doing this using what I call "subject affinities". A subject affinity is a declaration that a certain piece of text names or refers to a certain subject. To transform that into a link, or a connection, you then simply have to identify resources on that subject. But the identification of subjects are resources on subjects can be done dynamically and locally, which makes the taxonomy problem much more tractable. This approach decouples the semantics of the content itself from the semantics of the links, relationships, etc in the published content/data, allowing you to support multiple and emerging forms. 

I don't really work in this area anymore, but the full theory of it is explained in my book, Structured Writing: Rhetoric and Process, from XML Press. https://xmlpress.net/publications/structured-writing/

Not sure to what extent this addresses the original problem, but I suspect it might provide a way to decouple content from specific solutions to that problem, and simply how content is created for delivery where such solutions are applied. 

These two blog posts scratch the surface very very lightly: https://everypageispageone.com/2013/01/13/we-must-create-mutable-and-addressable-content/, https://everypageispageone.com/2011/08/01/more-links-less-time/ Remember that they were written for technical writers and content strategists, not for programmers.

**Comment by brimwats1** [comment]
I think that the Omeka S Value Suggest does the best job I've seen of this so far: https://omeka.org/s/modules/ValueSuggest/ 

It's a bit advanced for PubPub perhaps but it could be modified for simplicity. It is easy to add vocabularies and the autosuggest field pops up with terms added.

**Comment by brimwats1** [comment]
Another alternative is to take a look at how the Tropy project does their autosuggest field, but this is a local and desktop based platform so less relevant than Omeka I think

Thanks for the book recommendation @mbakeranalecta

**Comment by katycroffbell** [comment]
Yep, I already have links within the text between pubs. 

And funny, until this moment I have *only* considered what makes sense for my community, not machine readers.

**Reply by Dawit-Tegbaru** [reply]
Cool! I think you hit the sweet spot w/Basis and Related Material. I was a little worried about the additional Crossref declarations being hard to decipher so just wanted to give you the heads up. We'll look into adding these to the Connections list and get back to you w/updates asap.

**Comment by katycroffbell** [comment]
On a list “Basis” probably would have been confusing to me, but Dawit’s explanation ("Full results isBasisFor Summary”) makes perfect sense. If it isn’t already there, perhaps a link to the CrossRef page would be helpful.

**Comment by katycroffbell** [comment]
Hiya,
Are you able to add “Basis” & “Related Material" to the list???

Also, for some of my Connections (but not all), the authors are listed, not the title of the connected Pub. How can I make it be the title of the Pub? 

Thx!

**Reply by gabestein** [reply]
Hi Katy — we're working on an overhaul of how we register DOIs that will make it much easier for us to support the full range of connections. I don't know exactly when it will be available, but weeks not months, and it's on our list.

Re: connection authors vs. titles as I wrote over email, this is a legacy of how connections was built, originally to display reviews of content where displaying titles was redundant. Right now it displays the title if there are no authors, and authors if there are any present. We are planning to add a setting to allow admins to customize this display on a per-Pub basis, but I'm not sure yet when that will land.

**Comment by katycroffbell** [comment]
Hi Gabe,

Sounds great, keep me posted as new features become available! 

Katy

——————————————————
Katy Croff Bell, Ph.D.

President, Ocean Discovery League
National Geographic Explorer
